lib/matplotlib/tests/test_ticker.py

```diff
@@ -239,14 +240,120 @@ def test_set_params(self):
 
 
 class TestLogitLocator:
-    def test_set_params(self):
+    ref_basic_limits = [
+        (5e-2, 1 - 5e-2),
+        (5e-3, 1 - 5e-3),
+        (5e-4, 1 - 5e-4),
+        (5e-5, 1 - 5e-5),
+        (5e-6, 1 - 5e-6),
+        (5e-7, 1 - 5e-7),
+        (5e-8, 1 - 5e-8),
+        (5e-9, 1 - 5e-9),
+        (5e-10, 1 - 5e-10),
+    ]
+
+    ref_basic_major_ticks = [
+        1 / (10 ** np.arange(1, 3)),
+        1 / (10 ** np.arange(1, 4)),
+        1 / (10 ** np.arange(1, 5)),
+        1 / (10 ** np.arange(1, 6)),
+        1 / (10 ** np.arange(1, 7)),
+        1 / (10 ** np.arange(1, 8)),
+        1 / (10 ** np.arange(1, 9)),
+        1 / (10 ** np.arange(1, 10)),
+        1 / (10 ** np.arange(1, 11)),
+    ]
+
+    ref_maxn_limits = [(0.4, 0.6), (5e-2, 2e-1), (1 - 2e-1, 1 - 5e-2)]
+
+    @pytest.mark.parametrize(
+        "lims, expected_first_ticks",
+        zip(ref_basic_limits, ref_basic_major_ticks),
+    )
+    def test_basic_major(self, lims, expected_first_ticks):
         """
-        Create logit locator with default minor=False, and change it to
-        something else. See if change was successful. Should not exception.
+        Create logit locator with huge number of major, and tests ticks.
+        """
+        expected_ticks = np.sort(
+            np.concatenate(
+                (expected_first_ticks, (1 / 2,), 1 - expected_first_ticks)
```

Comment:
Something like
```
        expected = sorted([*expected_first, 1 / 2, *(1 - expected_first)])
```
would fit in a single line (it's not clear what "first" refers to, btw).

Reply:
Changed. (But it does'nt fit in one line (or maybe you expect I remove the `_ticks` in variable name, I prefer keeping a long name to have in mind that I manipulate ticks, it is not a strong preference).

"first" refers to ticks <.5, I build all the ticks using first ticks. I you have a better idea, let me know. I am not very happy with this name.

Reply:
low_ticks perhaps? at least low/high maps more clearly to <0.5/>0.5 than first/second, I think.
The exact formatting doesn't matter too much but it was just me once again not liking what black did :)

Reply:
Changed first to low.

Me too I do not like what black do, but what I like with black, it is that I can stop thinking of hundreds way to format the same thing.